src/main/scala/nl/knaw/dans/easy/stage/EasyStageDataset.scala

```
@@ -40,21 +41,40 @@ import scala.xml.NodeSeq
 
 object EasyStageDataset {
   val log: Logger = LoggerFactory.getLogger(getClass)
+  implicit val bagFactory = new BagFactory
 
   def main(args: Array[String]) {
     val props = new PropertiesConfiguration(new File(System.getProperty("app.home"), "cfg/application.properties"))
-    implicit val s = Settings(new Conf(args),props)
+    implicit val s = Settings(new Conf(args), props)
     run match {
       case Success(_) => log.info("Staging SUCCESS")
       case Failure(t) => log.error("Staging FAIL", t)
     }
   }
 
-  def run(implicit s: Settings): Try[(EasyMetadata, AdministrativeMetadata)] = {
+  // TODO: candidate for a possible dans-bagit-lib
+  /**
+   * Checks that all paths in `files` are part of the payload of the bag in `bagDir`. This means that they must be in at least one payload manifest.
+   *
+   * @param files the files to check
+   * @param bagDir the directory containing the bag
+   * @return Success if all files were part, otherwise Failure
+   */
+  def checkFilesInBag(files: Set[Path], bagDir: Path): Try[Unit] = {
```

Note, this function can also be called from `easy-ingest-flow` in the validate step, to fail fast if the mapping is not consistent with the files in the bag.

---

The [current] implementation in `easy-ingest-flow` gives more and better detailed error messages. To be reused it should be as complete and as thoroughly tested. Both components should have tests for their own test resources.

[current]: https://github.com/DANS-KNAW-jp/easy-ingest-flow/blob/1d68063fc97b9aba7d3a116b07eb8af8bf7bd321/src/main/scala/nl.knaw.dans.easy.ingestflow/flowsteps/validate/FlowStepValidate.scala#L167-L167

---

@jo-pol What more detail do you want? It returns the list of files that are in the set but not found in the bag.

---

Among others ingest-flow reports missing files in both directions, duplicate paths and more, therefore I linked somewhere into the validation method. All checks were added by yourself over time so why not implement them all here?

---

@jo-pol OK, I see were you're going with this now. Actually, I am not convinced that these validations should all be owned by `easy-stage-dataset`. I am more leaning towards factoring all validations out to a separate service. I'll leave this PR as is, to avoid scope creep, but I'll keep your remarks in mind. Something to discuss with @DANS-KNAW/easy, particularly @lindareijnhoudt.